arch/common/CMakeLists.txt

```diff
@@ -7,6 +7,10 @@ if(CONFIG_GEN_ISR_TABLES)
     CONFIG_GEN_ISR_TABLES
     sw_isr_common.c
     )
+  zephyr_library_sources_ifdef(
+    CONFIG_SHARED_INTERRUPTS
+    shared_irq.c
+    )
```

Comment:
There's some ambiguity, but this belongs in kernel/ not arch/common/ I'd say.  The arch layer is responsible for handling interrupts and dispatching to a _sw_isr_table[] entry, but everything downstream is OS behavior and belongs in the kernel.

Reply:
ACK. I'm assuming the same rationale is applicable to `sw_isr_common.c`, right?